README.md

```diff
@@ -0,0 +1,325 @@
+
+    $ pistachio # run mmcboot
+
+## TFTP Boot
+For TFTP boot, we need TFTP server serving kernel image (uImage), dtb (*.dtb)
+and initramfs filesystem.
+
+    openwrt$ sudo cp bin/pistachio/openwrt-pistachio-pistachio_marduk-uImage-initramfs /tftpboot/uImage
+    openwrt$ sudo cp bin/pistachio/pistachio_marduk.dtb /tftpboot
+
+### Setting up TFTP Server
+
+####Install tftpd package:
+
+    $ sudo apt-get install tftpd
+
+Create /etc/xinetd.d/tftp with following contents:
+    
+    service tftp
+    {
+     protocol        = udp
+     port            = 69
+     socket_type     = dgram
+     wait            = yes
+     user            = root
+     server          = /usr/sbin/in.tftpd
+     server_args     = /tftpboot
+     disable         = no
+    }
+
+Now restart service:
+
+    $ sudo service xinetd restart
+
+####Configure Uboot
+
+Now use Serial Console to connect device to host PC. Switch on device and press any key in first 2 seconds. After that you will drop to a u-boot console.
+
+To use tftp boot, set following environment variables.
+
+    # setenv ethaddr <xx:xx:xx:xx:xx:xx> # Set mac address for Ethernet
+(To generate mac address use command 
+
+    "echo "e2:"`od /dev/random -N5 -t x1 -An | sed 's/ /:/g' | cut -c1 --complement`"
+
+ on host machine)
+However beware of running out of IP addresses if we keep on switching the mac addresses.
+
+    # setenv serverip <server_ip> # Server IP address where TFTP and an NFS server is running
+    # saveenv # Save environment variables
+
+_note: No need to set these environment variables for next boot since these are already saved._
+
+Now start tftp boot:
+
+    # run ethboot
+
+
+You should see the logs on the console as below:
+
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  
+  BusyBox v1.24.1 (2016-03-04 17:43:09 IST) built-in shell (ash)
+
+  _______                     ________        __
+ |       |.-----.-----.-----.|  |  |  |.----.|  |_
+ |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
+ |_______||   __|_____|__|__||________||__|  |____|
+          |__| W I R E L E S S   F R E E D O M
+ -----------------------------------------------------
+ DESIGNATED DRIVER (Bleeding Edge, r48138)
+ -----------------------------------------------------
+  * 2 oz. Orange Juice         Combine all juices in a
+  * 2 oz. Pineapple Juice      tall glass filled with
+  * 2 oz. Grapefruit Juice     ice, stir well.
+  * 2 oz. Cranberry Juice
+
+$root@OpenWrt:/#
+$root@OpenWrt:/#
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can check "ifconfig -a" to check list of interfaces. Ethernet and 6loWPAN should be up, but WiFi is not yet configured.
+Note that 6loWPAN IP has been hardcoded to 2001:1418:0100::1/48.
+You can change that by editing /etc/config/network script and restarting it.
+
+    $ /etc/init.d/network restart
+
+
+### Known Issues:
+
+- Cleaned up kernel patches will be upstreamed soon.
+- WiFi is not configured in this build.
+- Booting from flash is not included.
```

Comment:
Booting from flash has been included but still needs to be documented. May be in different commit